homework/index.js

```diff
@@ -29,19 +31,120 @@
     });
     return elem;
   }
-
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  const root = document.getElementById('root');
+  function singlePageApplication(arr) {
+    arr.sort((a, b) => a.name.localeCompare(b.name));
+    // header
+    const div = createAndAppend('div', root, {
+      id: 'header-div',
+    });
+    createAndAppend('h1', div, { text: 'HYF Repositories', id: 'header' });
+    const selectMenu = createAndAppend('select', div, {
+      id: 'select-menu',
+    });
+    for (let i = 0; i < arr.length; i++) {
+      createAndAppend('option', selectMenu, {
+        text: arr[i].name,
+        value: i,
+      });
+    }
+    // right and left part
+    const containerDiv = createAndAppend('div', root, {
+      id: 'container',
+      class: 'container',
+    });
+    // right part
+    const firstTime = ['name', 'description', 'forks', 'updated_at'];
+    const rightDiv = createAndAppend('div', containerDiv, {
+      id: 'right-div',
+      class: 'contained',
+    });
+    const tHeads = ['Repository: ', 'Description: ', 'Forks: ', 'Update: '];
+    const table = createAndAppend('table', rightDiv, {
+      id: 'tableOfInformation',
+    });
+    for (let i = 0; i < tHeads.length; i++) {
+      const tr = createAndAppend('tr', table, {});
+      createAndAppend('th', tr, {
+        text: tHeads[i],
+      });
+      if (i === 0) {
+        const tableData = createAndAppend('td', tr, {
+          id: `tableData.${i}`,
+        });
+        createAndAppend('a', tableData, {
+          href: arr[i].html_url,
+          text: arr[i].name,
+          target: '_blank',
+        });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        createAndAppend('td', tr, {
+          text: arr[0][firstTime[i]],
+          id: `tableData.${i}`,
+        });
+      }
+    }
+    selectMenu.addEventListener('change', event => {
+      for (let i = 0; i < 4; i++) {
+        const changed = document.getElementById(`tableData.${i}`);
+        if (i === 0) {
+          changed.innerHTML = `<a target = "_blank" href = ${arr[event.target.value].html_url}> ${
```

Comment:
No need for spaces in between,
```suggestion
          changed.innerHTML = `<a target="_blank" href=${arr[event.target.value].html_url}> ${
```

and you may re-consider and try to avoid `.innerHTML` for a cleaner code and some other reasons as well, otherwise there is no need for any `.appendChild` any more :)